packages/esm-patient-registration-app/src/patient-registration/input/custom-input/identifier/identifier-input.component.tsx

```diff
@@ -111,27 +111,31 @@ export const IdentifierInput: React.FC<IdentifierInputProps> = ({ patientIdentif
       )}
       <div>
         {!patientIdentifier.required && patientIdentifier.initialValue && hideInputField && (
-          <Button
-            kind="ghost"
-            onClick={handleEdit}
-            iconDescription={t('editIdentifierTooltip', 'Edit')}
-            disabled={disabled}
-            hasIconOnly>
-            <Edit size={16} />
-          </Button>
+          <UserHasAccess privilege="Edit Patient Identifiers">
+            <Button
+              kind="ghost"
+              onClick={handleEdit}
+              iconDescription={t('editIdentifierTooltip', 'Edit')}
+              disabled={disabled}
+              hasIconOnly>
+              <Edit size={16} />
+            </Button>
+          </UserHasAccess>
         )}
         {initialValue && initialValue !== identifierValue && (
-          <Button
-            kind="ghost"
-            onClick={handleReset}
-            iconDescription={t('resetIdentifierTooltip', 'Reset')}
-            disabled={disabled}
-            hasIconOnly>
-            <Reset size={16} />
-          </Button>
+          <UserHasAccess privilege="Edit Patient Identifiers">
+            <Button
+              kind="ghost"
+              onClick={handleReset}
+              iconDescription={t('resetIdentifierTooltip', 'Reset')}
+              disabled={disabled}
+              hasIconOnly>
+              <Reset size={16} />
+            </Button>
+          </UserHasAccess>
         )}
         {!patientIdentifier.required && !defaultPatientIdentifierTypesMap[patientIdentifier.identifierTypeUuid] && (
-          <UserHasAccess privilege="coreapps.systemAdministration">
+          <UserHasAccess privilege="Delete Patient Identifiers">
```

Comment:
Users should probably be able to remove identifiers they added to this form with or without the "Delete Patient Identifier" privilege, i.e., if I'm creating a new patient or adding a new patient identifier to an existing patient, as long as that identifier hasn't been submitted to the backend, I should be able to remove it...

Reply:
But it'll only be added if the user has the privelege for "Add patient identifier", but yes, they should be able to delete the added identifier before creating the patient.

The better option here will be to only show this if the identifier was already present with the patient and the user has the privelege to delete the identifier.

Reply:
Sure thanks @ibacher @vasharma05 
so does it mean we can implement two buttons one for existing identifier and another for just created before submission?

Reply:
No, we can do this in this one only. We can go with this for now given the urgency